xbmc/cores/VideoPlayer/DVDSubtitles/SubtitleParserWebVTT.cpp

```diff
@@ -32,19 +32,36 @@ bool CSubtitleParserWebVTT::Open(CDVDStreamInfo& hints)
   if (!Initialize())
     return false;
 
+  CLog::LogF(LOGINFO, "Initialized WebVTT Parser");
+
   CWebVTTHandler m_webvttHandler;
 
   if (!m_webvttHandler.Initialize())
     return false;
 
-  // Get the first chars to check WebVTT signature
-  if (!m_webvttHandler.CheckSignature(m_pStream->Read(10)))
+  CLog::LogF(LOGINFO, "Initialized WebVTT Handler");
+
+  std::string line;
+
+  // WebVTT signature is according to spec length-variable and stops
+  // with two consecutive line terminators.
+  // Get the first line to check WebVTT signature.
+  if(!(m_pStream->ReadLine(line) && m_webvttHandler.CheckSignature(line)))
+  {
+    return false;
+  }
+
+  //Check if we have a second empty line as necessitated by spec.
+  //More empty lines might follow but will be handled by line decode.
+  if(!(m_pStream->ReadLine(line) && line.empty()))
+  {
     return false;
-  m_pStream->Seek(0);
+  }
+
+  CLog::LogF(LOGINFO, "Matched valid WebVTT signature");
```

Comment:
not needed, as comment above, if the signature fails the signature check method must print an error or silently continue